add-on/src/lib/ipfs-request.js

```diff
@@ -144,7 +144,11 @@ export function createRequestModifier (getState, dnslinkResolver, ipfsPathValida
 
       // When local IPFS node is unreachable , show recovery page where user can redirect
       // to public gateway.
-      if (!state.nodeActive && request.type === 'main_frame' && sameGateway(request.url, state.gwURL)) {
+      if (!state.nodeActive && //                  node is not active
+        !state.redirect && //                      this is not a redirect request
```

Comment:
nit: this is a global flag that controls if companion is redirecting requests to local gateway, or not.
This was not fixing the problem, and the comment was only confusing the reader – I've restored old version and applied proper fix in `sameGateway`.